src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/locationfinder/AppointmentLocationsService.kt

```diff
@@ -46,6 +53,26 @@ class AppointmentLocationsService(
       .getScheduledAppointments(specification.agencyId, specification.date)
       .filter { it.endTime != null }
 
+  fun findVideoLinkBookingOptions(specification: VideoLinkBookingSearchSpecification): VideoLinkBookingOptions {
+    val excludedAppointmentIds =
+      specification.vlbIdToExclude?.let { appointmentIdsFromBookingIds(listOf(it)) } ?: emptyList()
+
+    val locationIds = setOfNotNull(
+      specification.preAppointment?.locationId,
+      specification.mainAppointment.locationId,
+      specification.postAppointment?.locationId
+    )
+
+    val scheduledAppointments =
+      prisonApiService
+        .getScheduledAppointments(specification.agencyId, specification.date).asSequence()
```

Comment:
Just wondering what the rationale was for using Sequences here?

Reply:
To avoid creating the intermediate collections that would arise from using a list or similar.